<?xml version="1.0"?>
  <launch>


    <node pkg="env_detection_core" type="env_detection_core" name="eds_core_node" output="screen" respawn="false" respawn_delay="0">
        <param name="input_map_topic" type="string" value="/map" />
        <param name="env_value_topic" type="string" value="/eds/env_value" />
        <param name="set_layer_topic" type="string" value="/eds/set_layer" />
    </node>

    <node pkg="env_detection_core" type="odometry_node.py" name="eds_odometry_node" output="log" respawn="false" respawn_delay="0" />
        
    <node pkg="env_detection_sensors" type="env_detection_sensors" name="sensors_node" output="log" respawn="false" respawn_delay="0">
        <param name="sensors_topic" type="string" value="/sensors" />
        <param name="env_value_topic" type="string" value="/eds/env_value" />
    </node>

    <include file="$(find env_detection_core)/launch/gmapping.launch" />
   
    <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen">
      <rosparam command="load" file="$(find env_detection_core)/config/parameters.yaml" />
    </node>
  <!-- Launch RViz with the demo configuration -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find env_detection_core)/rviz/eds_rosbag1.rviz" />     
</launch>
  
